src/fastoad/io/configuration/configuration.py

```diff
@@ -414,33 +430,39 @@ def _add_design_vars(self, model, auto_scaling):
                 design_var_table["ref"] = design_var_table["upper"]
             model.add_design_var(**design_var_table)
 
-    def _get_problem_inputs(self, problem: FASTOADProblem) -> Tuple[om.IndepVarComp, VariableList]:
+    def _get_problem_inputs(
+        self, problem: FASTOADProblem, ivc_output_format=True
+    ) -> Tuple[om.IndepVarComp, VariableList]:
         """
-        Reads input file for the configure problem.
+        Reads input file for the configured problem.
 
         Needed variables are returned as an IndepVarComp instance while unused variables are
         returned as a VariableList instance.
 
         :param problem: problem with missing inputs. setup() must have been run.
         :return: IVC of needed input variables, VariableList with unused variables.
         """
-        mandatory, optional = get_unconnected_input_names(problem, promoted_names=True)
-        needed_variable_names = mandatory + optional
+        # TODO: shift this to from_problem
+        problem_copy = copy.deepcopy(problem)
+        problem_variables = VariableList().from_problem(problem_copy)
+        problem_inputs_names = [var.name for var in problem_variables if var.is_input]
```

Comment:
Here also, this line will be avoided if `from_problem` does the job of providing only inputs.